topic-operator/src/main/java/io/strimzi/operator/topic/zk/ZkImpl.java

```diff
@@ -8,382 +8,229 @@
-        return this;
-    }
-
-    @Override
-    public Zk exists(String path, Handler<AsyncResult<Stat>> handler) {
-        ZooKeeper zookeeper;
-        synchronized (this) {
-            zookeeper = zk;
-        }
-        if (zookeeper == null) {
-            handler.handle(Future.failedFuture(new IllegalStateException("Not connected")));
-            return this;
-        }
-        final AsyncCallback.StatCallback callback = (rc, path1, ctx1, stat) -> {
-            Watcher.Event.EventType eventType = (Watcher.Event.EventType) ctx1;
-            if (eventType == null // first time
-                    || eventType == Watcher.Event.EventType.NodeCreated
-                    || eventType == Watcher.Event.EventType.NodeDeleted
-                    || KeeperException.Code.get(rc) != KeeperException.Code.OK) {
-                Future<Stat> future = mapResult(path1, rc, stat);
-                vertx.runOnContext(ignored -> {
-                    final Handler<AsyncResult<Stat>> watch = getExistsWatchHandler(path);
-                    if (eventType != null && watch != null) {
-                        // Only call the handlers if callback fired due to watch
-                        watch.handle(future);
-                    }
-                    if (eventType == null && handler != null) {
-                        handler.handle(future);
-                    }
-                });
-            }
-        };
-        final Watcher watcher;
-        if (getExistsWatchHandler(path) != null) {
-            watcher = new Watcher() {
-                @Override
-                public void process(WatchedEvent event) {
-                    if (getExistsWatchHandler(path) != null) {
-                        // Reset the watch if there still is a handler
-                        zookeeper.exists(path, this,
-                                callback, event.getType());
+        workerPool().executeBlocking(
+            future -> {
+                try {
+                    if (zookeeper.delete(path, version)) {
+                        future.complete();
+                    } else {
+                        future.fail(new ZkNoNodeException());
                     }
+                } catch (Throwable t) {
+                    future.fail(t);
                 }
-            };
-        } else {
-            watcher = null;
-        }
-        zookeeper.exists(path, watcher, callback, null);
-        return this;
-    }
-
-    @Override
-    public Zk watchExists(String path, Handler<AsyncResult<Stat>> watcher) {
-        watches.put(PREFIX_EXISTS + path, watcher);
+            },
+            handler);
         return this;
     }
 
-    @Override
-    public Zk unwatchExists(String path) {
-        watches.remove(PREFIX_EXISTS + path);
-        return this;
+    private WorkerExecutor workerPool() {
+        return vertx.createSharedWorkerExecutor(getClass().getName(), 4);
```

Comment:
what is the rationale for 4?

Reply:
What do you think is the right number of threads in the pool? I think 4 is probably _way_ more than needed. 1 would probably suffice, or maybe 2. Wdyt?

Reply:
It could depend on how many calls you need to start in parallel having free threads from the pool. For this reason, I was asking if there were a rationale behind it. If not, then 4 is a reasonable choice.